checkov/common/graph/graph_builder/utils.py

```diff
@@ -37,6 +37,6 @@ def run_function_multithreaded(
 def filter_sub_keys(key_list: list[str]) -> list[str]:
     filtered_key_list = []
     for key in key_list:
-        if not any(other_key != key and other_key.startswith(key) for other_key in key_list):
+        if not any(other_key != key and other_key.startswith(key) for other_key in key_list) and f"dynamic.{key.split('.')[0]}" not in key_list:
```

Comment:
This condition is getting pretty big, can we refactor it into a function to explain what it stands for?